src/validators/admin.validator.ts

```diff
@@ -0,0 +1,48 @@
+import { NextFunction } from 'express';
+import { IRequest, IResponse } from '../interfaces/vendor';
+import { validator } from '../utils';
+import * as regexes from '../utils/constants/regexes';
+import { ValidationException } from '../utils/exceptions';
+
+const validateAdminRegistration = (req: IRequest, res: IResponse, next: NextFunction) => {
+    let errors = {};
+
+    // Trim the inputs
+    req.body = validator.trimReqBody(req.body);
+
+    // name validation
+    if (validator.isEmpty(req.body.name)) {
+        errors = { ...errors, name: 'Name is required' };
+    } else if (!regexes.NAME_REGEX.test(req.body.name)) {
+        errors = { ...errors, name: 'Name is invalid' };
+    } else if (req.body.name.length < 3 || req.body.name.length > 255) {
```

Comment:
why not move 3 and 255 to constants and use it in user model and here so that there will only be a single source ?

Reply:
- User model will be removed further and validation will only done once here
- There will be no redundant validation
- For now, please ignore Mongoose related code